test/unit/features/options/template.spec.js

```diff
@@ -88,4 +88,13 @@ describe('Options template', () => {
       `avoid using JavaScript unary operator as property name: "delete()" in expression @click="delete('Delete')"`
     ).toHaveBeenWarned()
   })
+
+  it('warn error in generated function (v-on)', () => {
```

Comment:
The spec name imo should be something like,
"should not warn dot prefixed keywords".

As the warning would appear even on general JS objects with a `delete` property and not just `$http`.